pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -2781,11 +2894,26 @@ func (c *Cloud) EnsureLoadBalancerDeleted(clusterName string, service *api.Servi
 
 		// Collect the security groups to delete
 		securityGroupIDs := map[string]struct{}{}
+
+		// shared securitygroup ID for reference
+		var ssgID string
+		var err error
+		if s.cfg.Global.EnableSharedSecurityGroupIngress {
```

Comment:
@justinsb Could you elaborate why this case is not reachable?? When `EnsureLoadBalancerDeleted` is called, this case is always accessed?